### DISCUSSION TITLE
Running the Demo on my own environment

### DISCUSSION THREAD

**Discussion by johnnytranamp** [caplin/FlexLayout#356] [General]
I've git cloned the whole repo onto my desktop and want to play around with the demo on my personal workstation as I'm a fairly new developer still trying to learn all it's features and code. I haven't had success in creating a live server for it as every time I try to run npm run start / npm run build:demo it returns the error message below. Any help would be a greatly appreciated! (I see that there's already 3 webpack files that came along with the files but I can't seem to figure it out)

'webpack' is not recognized as an internal or external command,
operable program or batch file.

**Comment by edemaine** [comment]
It sounds like you need to run `npm install` (if you haven't already). This is a common first step for Node projects.

**Reply by johnnytranamp** [reply]
This is what I ran into when trying to npm install

PS C:\Users\JohnnyTran\Desktop\FlexLayout> npm i 
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: flexlayout-react@0.7.5
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   peer react@"^18.0.0" from the root project
npm ERR!   peer react@"^18.2.0" from next@13.0.5
npm ERR!   node_modules/next
npm ERR!     peerOptional next@">=8" from @cypress/react@5.12.4
npm ERR!     node_modules/@cypress/react
npm ERR!       dev @cypress/react@"^5.12.4" from the root project
npm ERR!   1 more (react-dom)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^=16.x || ^=17.x" from @cypress/react@5.12.4
npm ERR! node_modules/@cypress/react
npm ERR!   dev @cypress/react@"^5.12.4" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!

**Reply by edemaine** [reply]
Try `npm i --legacy-peer-deps` or `npm i --force`.

**Reply by johnnytranamp** [reply]
I was able to receive this message in the terminal after I `npm i --force`  but there wasn't an available localhost for me to open up. This was the new message in the terminal.

PS C:\Users\JohnnyTran\Desktop\FlexLayout> npm run build:demo

> flexlayout-react@0.7.5 build:demo
> webpack

asset ./bundles/demo.js 1.51 MiB [emitted] (name: demo) 1 related asset
orphan modules 15.9 KiB [orphan] 10 modules
runtime modules 1.25 KiB 6 modules
cacheable modules 1.42 MiB
  modules by path ./src/ 230 KiB
    modules by path ./src/view/*.tsx 96.2 KiB 15 modules
    modules by path ./src/model/*.ts 99.7 KiB 15 modules
    modules by path ./src/*.ts 31.2 KiB 10 modules
    ./src/PopupMenu.tsx 3.01 KiB [built] [code generated]
  modules by path ./node_modules/ 1.16 MiB 23 modules
  modules by path ./examples/demo/*.tsx 37.7 KiB
    ./examples/demo/App.tsx 24.5 KiB [built] [code generated]
    ./examples/demo/NewFeatures.tsx 2.33 KiB [built] [code generated]
    ./examples/demo/PopupMenu.tsx 2.25 KiB [built] [code generated]
    ./examples/demo/TabStorage.tsx 7.5 KiB [built] [code generated]
    ./examples/demo/Utils.tsx 1.14 KiB [built] [code generated]
webpack 5.72.0 compiled successfully in 3817 ms

**Reply by edemaine** [reply]
`npm start` (not `npm run build:demo`) is what will run a server on localhost to connect to.

**Reply by johnnytranamp** [reply]
Its running now thank you!
